picklists/views.py

```
@@ -86,6 +88,23 @@ def get(self, request, **kwargs):
         return form_page(request, self.form, data=self.picklist_item)
 
     def post(self, request, **kwargs):
+        # Letter paragraphs are passed through the Django template engine, so we need
+        # to make sure they're valid.
+        if request.POST.get("type") == "letter_paragraph":
+            template_engine = template.engine.Engine.get_default()
+            try:
+                template_engine.from_string(request.POST["text"])
```

Potential keyerror if submitted with empty `text`?

Otherwise lg2m :)

---

@a-bek Figure it's better to raise a `KeyError` than to silently accept a post with missing text.